src/battle_ai_main.c

```diff
@@ -2638,6 +2638,27 @@ static s16 AI_CheckBadMove(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
              && AI_WhoStrikesFirst(battlerAtk, battlerDef, move) == AI_IS_SLOWER)
                 score -= 10;
             break;
+        case EFFECT_JUNGLE_HEALING:
+            if (AtMaxHp(battlerAtk)
+             && AtMaxHp(BATTLE_PARTNER(battlerAtk))
+             && !(gBattleMons[battlerAtk].status1 & STATUS1_ANY)
+             && !(gBattleMons[BATTLE_PARTNER(battlerAtk)].status1 & STATUS1_ANY))
+                score -= 10;
+            else if (gBattleMons[battlerAtk].status1 & STATUS1_ANY
+                  || gBattleMons[BATTLE_PARTNER(battlerAtk)].status1 & STATUS1_ANY
+                  || AI_DATA->hpPercents[battlerAtk] >= 90
+                  || AI_DATA->hpPercents[BATTLE_PARTNER(battlerAtk)] >= 90)
+                score -= 9; // No point in healing, but should at least do it if there's nothing better or if it's afflicted by a status ailment.
```

Comment:
I'm getting a sense I already commented something like this before, but this logic is wrong. Going line by line:

`gBattleMons[battlerAtk].status1 & STATUS1_ANY`: This is saying if the attacker has any status condition do score -9. This completely defeats the purpose of the move since the attacker will basically try to never heal its own status condition.

`gBattleMons[BATTLE_PARTNER(battlerAtk)].status1 & STATUS1_ANY`: Same as above except now if the partner has a status condition the AI won't use the move.

`AI_DATA->hpPercents[battlerAtk] >= 90`: So what happens if the partner is at 1 HP but the attacker is at full health? The AI wouldn't use this move because one of the mons on the field is already near max HP.

`AI_DATA->hpPercents[BATTLE_PARTNER(battlerAtk)] >= 90`: Same as above except now the AI won't heal itself if its partner is near max HP.

Reply:
This is located in a function that demotivates the AI from using the move in question if any of the conditions are met.

After the initial `if` right above which makes sure to completely demotivate the AI from using the move if they're at full HP or if they don't have any status ailments, the AI would be a bit more inclined to use the move if they did have any status ailment or if they lacked a bit of HP *(which I should replace with a `!AtMaxHp(battlerAtk)` check and an equivalent for their partner probably.)*

Reply:
> This is located in a function that demotivates the AI from using the move in question if any of the conditions are met

Yes, but let me explain how you're supposed to use `AI_CheckBadMove`. This function is supposed to tell the AI when it's a _bad idea_ to do something. As in, the move won't work so the AI shouldn't use it. You've already covered the case where none of the conditions needed for Jungle Healing work (in the condition with score - 10), so fundamentally it is not a "bad idea" to use the move anymore once it's been determined it _will work_. Only `AI_CheckViability` is supposed to be used for "smart AI logic". If you don't want the AI using the move because there would be better options, you just don't give a plus score there.

> There, the code for Jungle Healing's effect as you can see it in this PR right now is the following

Yes, and look what happens in this case, for instance. Say the attacker has a status condition, is at low HP, and the partner also has a status condition (but is at full HP). That's `+3`. But oh, wait, in `AI_CheckBadMove` you have `gBattleMons[battlerAtk].status1 & STATUS1_ANY` gives `-9`. That's a net `-6` and the AI won't use the move (not to mention you also have `AI_DATA->hpPercents[BATTLE_PARTNER(battlerAtk)] >= 90` so that would cause a `-9` as well). Do you see what the problem is now?

Reply:
Yeah, I think I do. Thanks. I'll remove the -9 check completely.

In regards to the original -10 check, what do you think about a change like this?

```diff
        case EFFECT_JUNGLE_HEALING:
-           if (AtMaxHp(battlerAtk)
-            && AtMaxHp(BATTLE_PARTNER(battlerAtk))
-            && !(gBattleMons[battlerAtk].status1 & STATUS1_ANY)
-            && !(gBattleMons[BATTLE_PARTNER(battlerAtk)].status1 & STATUS1_ANY))
+           if ((AtMaxHp(battlerAtk) && AtMaxHp(BATTLE_PARTNER(battlerAtk)))
+            || (!(gBattleMons[battlerAtk].status1 & STATUS1_ANY) && !(gBattleMons[BATTLE_PARTNER(battlerAtk)].status1 & STATUS1_ANY)))
                score -= 10;
            break;
```

Reply:
Leave what you had before; it was good. That's now saying if they're both at full HP or both don't have status conditions then don't use the move. You don't want that because in the case where they're both at 1 HP but still both don't have status conditions, the AI will choose not to use the move.